src/components/SearchFetch/SearchFetch.tsx

```diff
@@ -4,37 +4,96 @@ import * as queryResult from 'avenger/lib/QueryResult';
 import View from '../View/View';
 import { WithQueries } from 'avenger/lib/react';
 import { searchResturantQuery } from '../../queries/queries';
-import { FormattedMessage } from 'react-intl';
+import { FormattedMessage, useIntl } from 'react-intl';
 import { SearchResults } from '../SearchResults/SearchResults';
 import { ErrorMsg } from '../ErrorMsg/ErrorMsg';
-import { LoadingSpinner } from '../LoadingSpinner/LoadingSpinner';
+import LoadingSpinner from '../LoadingSpinner/LoadingSpinner';
+import Button from '../Button/Button';
+import NavBar from '../NavBar/NavBar';
+import ScrollView from '../ScrollView/ScrollView';
+import './searchfetch.scss';
+import { PAGE_SIZE_CONFIG } from '../../API/API';
 
 export function SearchFetch(props: { location: string; range: number }) {
-  const provideLoader = () => <LoadingSpinner />;
+  const [curPage, setPage] = React.useState<{ page: number }>({
+    page: 0
+  });
+
+  const intl = useIntl();
+  const nextLabel = intl.formatMessage({ id: 'SearchContainer.result.next.button.label' });
+  const prevLabel = intl.formatMessage({ id: 'SearchContainer.result.prev.button.label' });
+
+  const curPageIsLast = (page: number, total: number) =>
+    page < Math.ceil(total / PAGE_SIZE_CONFIG) - 1;
+
+  const prevPage = (page: number) =>
+    page > 0 ? setPage({ page: page - 1 }) : setPage({ page: 0 });
+
+  const nextPage = (page: number, total: number) => {
+    if (curPageIsLast(page, total)) {
+      setPage({ page: page + 1 });
+    }
+  };
+
+  const renderLoader = () => (
+    <div style={{ position: 'relative', height: 200 }}>
+      <LoadingSpinner size={150} />
+    </div>
+  );
+
+  const renderPrevButton = () =>
+    !curPage.page ? '' : <Button onClick={() => prevPage(curPage.page)} label={prevLabel} />;
+
+  const renderNextButton = (total: number) =>
+    curPageIsLast(curPage.page, total) ? (
+      <Button label={nextLabel} onClick={() => nextPage(curPage.page, total)} />
+    ) : (
+      ''
```

Comment:
Please, don't render empty strings instead of components. You can both return `null` or write this as:
```suggestion
    curPageIsLast(curPage.page, total) && (
       <Button label={nextLabel} onClick={() => nextPage(curPage.page, total)} />
    )
```